src/ui/components/search/searchcomponent.js

```diff
@@ -487,7 +498,8 @@ export default class SearchComponent extends Component {
     // serialized and submitted.
     if (typeof this.redirectUrl === 'string') {
       if (this._allowEmptySearch || query) {
-        window.location.href = this.redirectUrl + '?' + params.toString();
+        const newUrl = this.redirectUrl + '?' + params.toString();
+        window.open(newUrl, this.redirectUrlTarget) || (window.location.href = newUrl);
```

Comment:
For my own edification, what's the `|| (window.location.href = newUrl)` do? If the `target` is `_self` does `window.open` return false?

Reply:
Users can configure their browser settings so that window.open won't work well. For example, if someone has an aggressive pop-up blocking setup, this would fall back to opening the redirect URL in the current frame using the `window.location.href = newUrl`. To my knowledge, _self should work OK with default browser settings.